src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy strategy;
+
+    public ShopServiceImpl(OperationStrategy strategy) {
+        this.strategy = strategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        transactions.forEach(t -> strategy.get(t.getOperation()).applyOperation(t));
+    }
+}
```

Comment:
The class `ShopServiceImpl` correctly uses the Strategy pattern to delegate the operation logic. However, to strictly adhere to the SOLID principles, consider validating the `transactions` list before processing it to ensure that it is not null or empty. This is to prevent `NullPointerException` or unnecessary processing. You can throw a custom exception if the input is invalid.